packages/hash/api/src/model/block.model.ts

```diff
@@ -123,6 +123,48 @@ class __Block extends Entity {
 
     return firstLink.getDestination(client);
   }
+
+  async updateBlockData(
+    client: DbClient,
+    params: { updatedByAccountId: string; updatedDataEntity: Entity },
+  ): Promise<Entity> {
+    const { updatedByAccountId, updatedDataEntity } = params;
+    const blockDataLinks = await this.getOutgoingLinks(client, {
+      path: ["data"],
+    });
+
+    const [previousBlockDataLink, ...otherLinks] = blockDataLinks;
+
+    if (!previousBlockDataLink) {
+      throw new Error(
+        `Critical: block with entityId ${this.entityId} in account with accountId ${this.accountId} has no linked block data entity`,
+      );
+    }
+    if (otherLinks.length > 0) {
+      throw new Error(
+        `Critical: block with entityId ${this.entityId} in account with accountId ${this.accountId} has more than one linked block data entity`,
+      );
+    }
+
+    if (
+      previousBlockDataLink.destinationEntityId === updatedDataEntity.entityId
+    ) {
+      const previousBlockData = await this.getBlockData(client);
+      return previousBlockData;
+    }
+
+    await previousBlockDataLink.delete(client, {
+      deletedByAccountId: updatedByAccountId,
+    });
+
+    await this.createOutgoingLink(client, {
+      destination: updatedDataEntity,
+      createdByAccountId: updatedByAccountId,
+      stringifiedPath: "$.data",
+    });
+
+    return updatedDataEntity;
```

Comment:
The major thing it's meant to do is handle deleting the existing link & creating a new link once the block data has been swapped. If the block data wasn't swapped, it returns the current block data. I'd probably rename the method to `swapBlockData`, so it's a bit clear... Since it isn't meant to handle updates to the current block data's properties. Will refactor and get back to you

Reply:
Refactored here => https://github.com/hashintel/hash/pull/506/commits/e54687ecee0f12783a8fc5d3a40eec443c6e49bc.